### DISCUSSION TITLE
How to initialize Tree component with initialCheckedState?

### DISCUSSION THREAD

**Discussion by comertcimen** [mantinedev/mantine#6641] [Q&A] [answered]
It is possible to use useTree with initialSelectedState with node values but not for checked state.
Is it possible to extend it as the example?

```ts
interface UseTreeInput {
  ...rest

  /** Initial checked state of nodes */
  initialCheckedState?: string[];
}
```

**Comment by rtivital** [comment, marked as answer]
Currently, it is not supported, it might be added in future releases

**Comment by fredgig** [comment]
This is a limitation I came across recently. Seems like an oversight since we already have initialSelectedState and initialExpandedState.
